relay-server/src/actors/processor.rs

```diff
@@ -2439,6 +2439,94 @@ mod tests {
+            let mut scopes = Vec::<(Scoping, ProjectMetrics)>::new();
+
+            let quota = Quota {
+                id: Some("testing".into()),
+                categories: vec![DataCategory::MetricBucket].into(),
+                scope: relay_quotas::QuotaScope::Organization,
+                scope_id: Some("1".into()), // we will ratelimit buckets with scope org-id == 1
+                limit: Some(0),
+                window: None,
+                reason_code: Some(ReasonCode::new("test")),
+            };
+
+            let project_state = {
+                let mut x = ProjectConfig::default();
+                x.quotas.push(quota);
+                let mut project_state = ProjectState::allowed();
+                project_state.config = x;
+                Arc::new(project_state)
+            };
+
+            scopes.push((
+                Scoping {
+                    organization_id: 1, // will be ratelimited
+                    project_id: ProjectId::new(21),
+                    project_key: ProjectKey::parse("a94ae32be2584e0bbd7a4cbb95971fee").unwrap(),
+                    key_id: Some(17),
+                },
+                ProjectMetrics {
+                    buckets: vec![Bucket {
+                        name: "d:transactions/foo".to_string(),
+                        value: BucketValue::Counter(1.0),
+                        timestamp: UnixTimestamp::now(),
+                        tags: Default::default(),
+                        width: 10,
+                    }],
+                    project_state: project_state.clone(),
+                },
+            ));
+
+            scopes.push((
+                Scoping {
+                    organization_id: 0, // will not be ratelimited
+                    project_id: ProjectId::new(21),
+                    project_key: ProjectKey::parse("a94ae32be2584e0bbd7a4cbb95971fee").unwrap(),
+                    key_id: Some(17),
+                },
+                ProjectMetrics {
+                    buckets: vec![Bucket {
+                        name: "d:transactions/bar".to_string(),
+                        value: BucketValue::Counter(1.0),
+                        timestamp: UnixTimestamp::now(),
+                        tags: Default::default(),
+                        width: 10,
+                    }],
+                    project_state,
+                },
+            ));
+
+            EncodeMetrics { scopes }
+        };
+
+        let received_batches = Arc::new(AtomicUsize::new(0));
+        let f = |x: &mut Arc<AtomicUsize>, _: Store| {
+            x.fetch_add(1, SeqCst); // counts the number of not-ratelimited batches.
+        };
+
+        let (store, handle) = mock_service("store_forwarder", received_batches.clone(), f);
+        create_test_processor(Default::default()).encode_metrics_processing(message, &store);
+        drop(store);
+        handle.await.unwrap();
+
+        let received_batches = received_batches.load(SeqCst);
```

Comment:
For extra safety: could we validate that the batch we get here is the expected above (i.e. the one with org id 0)?

Reply:
done ^^